test/e2e/instrumentation/monitoring/custom_metrics_deployments.go

```diff
@@ -250,6 +258,19 @@ func CreateAdapter(adapterDeploymentFile string) error {
 	return err
 }
 
+func createClusterAdminBinding() error {
+	stdout, stderr, err := framework.RunCmd("gcloud", "config", "get-value", "core/account")
+	if err != nil {
+		framework.Logf(stderr)
+		return err
+	}
+	serviceAccount := strings.TrimSpace(stdout)
+	framework.Logf("current service account: %q", serviceAccount)
+	stat, err := framework.RunKubectl("create", "clusterrolebinding", ClusterAdminBinding, "--clusterrole=cluster-admin", "--user="+serviceAccount)
```

Comment:
where are you using this role binding? looks like you're granting the cluster-admin ClusterRole to the gcloud service account name. I think Kubernetes SA and GCP SA are different concepts. So not sure what kinds of permissions you want and what you want to grant it to, but this looks suspicious to me.

Reply:
I'm later creating a clusterrole, for example the one present in: https://github.com/GoogleCloudPlatform/k8s-stackdriver/blob/master/custom-metrics-stackdriver-adapter/deploy/staging/adapter_old_resource_model.yaml
(cluster role named "external-metrics-reader")

Following GCE documentation, I need to create this binding in order to create any cluser role, see https://cloud.google.com/kubernetes-engine/docs/how-to/role-based-access-control (section "Prerequisites for using Role-Based Access Control). I don't understand fully why this step is necessary on GKE, while it just works fine on GCE.